### PR TITLE
Fix bug in boost property tree doesKeyExist() method

### PR DESCRIPTION
**Pull request by uschwes**

### CODE REVIEW

**Comment by uschwes** [comment]
Waiting for an lgtm ☺

File: sm_property_tree/src/BoostPropertyTreeImplementation.cpp [hunk 1]
```diff
@@ -91,7 +91,8 @@ namespace sm {
 
   bool BoostPropertyTreeImplementation::doesKeyExist(const std::string & key) const
   {
-    boost::optional<std::string> val = _ptree.get<std::string>(boost::property_tree::ptree::path_type(key.c_str(),'/'));
+    std::string keyNoLeadingSlash = key[0] == '/' ? key.substr(1) : key; // Note: _ptree.get_optional does not seem to like leading slashes
```
Comment (HannesSommer) [inline sm_property_tree/src/BoostPropertyTreeImplementation.cpp]: This does lack a check against empty.

**Comment by uschwes** [comment]
There are some methods like `loadXml()` (or `saveXml()` actually in the first place) that fail with empty keys. However the default `set` and `get` methods work.
Question would be whether we accept empty keys or not.
If we do support it I would say that `doesKeyExist("")` returns false iff the tree is empty.

**Comment by uschwes** [comment]
No, the last suggestion is not good. I changed now the behavior so that `doesKeyExist` always returns true for an empty key. The empty key always refers to the subtree starting at `/`.

**Comment by HannesSommer** [comment]
Yes, I agree with true for empty key.
Great. Lgtm then.

**Pull request merged**